lib/config/parameters_parser.js

```diff
@@ -72,31 +79,38 @@ export class ParametersParser {
     params.splice(languageParamIndex, 2);
   }
 
-  static sanitizeDirectoryParamOptions(params, directoryParamIndex) {
-    const directoryOption = params[directoryParamIndex + 1];
-    if (isUndefined(directoryOption)) {
-      throw new ConfigurationParsingError('Must send a route for --directory option');
+  static sanitizeParamWithArgument(params, paramWithArgumentIndex, paramName) {
+    const paramOption = params[paramWithArgumentIndex + 1];
+    if (isUndefined(paramOption)) {
+      throw new ConfigurationParsingError(`Must send a route for ${paramName} option`);
     }
-    const directoryConfig = [`-d ${directoryOption}`];
-    this.removeParameterAtIndex(params, directoryParamIndex);
-    return [...params, ...directoryConfig];
+    const paramConfig = [`${paramName} ${paramOption}`];
+    this.removeParameterAtIndex(params, paramWithArgumentIndex);
+    return [...params, ...paramConfig];
   }
 
   static validateConfigurationParams(paramsList) {
     paramsList.forEach((param, index) => {
-      if (this.isLanguageParam(param)) {
+      if (this.isParamWithArgument(param, ['-l', '--language'])) {
         this.validateLanguageOption(paramsList, index);
       }
 
-      const previousParam = paramsList[index - 1];
-      const previousParamIsDirectoryFlag = previousParam === '-d' || previousParam === '--directory';
-
-      if (!this.isRawConfigurationParam(param) && !previousParamIsDirectoryFlag) {
-        throw new ConfigurationParsingError('Run configuration parameters should always be sent at the end of test paths routes');
-      }
+      this.validateIsNotPathParam(param, paramsList, index);
     });
   }
 
+  static validateIsNotPathParam(param, paramsList, paramIndex) {
+    const previousParam = paramsList[paramIndex - 1];
+    this.assertParamIsNotPathParam(param, previousParam);
+  }
+
+  static assertParamIsNotPathParam(param, previousParam) {
+    const previousParamIsDirectoryOrExtensionFlag = ['-d', '--directory', '-e', '--extension'].includes(previousParam);
```

Comment:
Let's extract `-d`, `--directory` and similar strings to constants, because they are used more than once in the code